Lua patch: revamping installing Lua Workout

junoocha opened the pull request:
This workout was heavily revamped due to the many confusions users had with actually downloading Lua and using it on their computer.

As such, I considered an alternative for users to easily code on Lua. I realized that most people are here to be able to code on Roblox as that is the most popular platform for Lua coding and as such, downloading Lua is unnecessary as the Lua compiler is built into Roblox Studio. Thus, simply having an online IDE would be sufficient for following along in this course. 

For users that do want to install Lua, I did make some clarifications with installing Lua and suggested that for any specific issues, watching tutorials would help, as its pretty difficult for new users to make the changes on their computer and coding environment. I did make a notice on insight 2 that insights 2-4 can be skipped if the user just plans to code on Roblox.

Additionally, I found an android app that also served as a Lua ide, allowing users to code Lua on their phones. A lot of people were asking for a mobile IDE to play around with Lua on there, and as I know that the current Enki app doesn't support Lua code, this app may serve as an alternative to coding on the go. I unfortunately don't have access to any apple products, but I do know of some Lua Ides on the apple store. I haven't put them in yet as I have not used them. I can put them in with further review.

On a side note, with this changes, a note may have to be added for input/output operations, as it does involve writing and exporting files and cannot be done on an online IDE.

enki-hq commented:
### Insight previews:
- [ ] [enkidevs/curriculum/lua/lua-core/intro-to-functions-in-lua-iii/vararg-expression-in-lua-ii.md](https://app.enki.com/public/preview-insight/enkidevs/curriculum/lua/lua-core/intro-to-functions-in-lua-iii/vararg-expression-in-lua-ii.md?ref=885261a0c1b5c73d2c6127d272009604b479b7f5)
- [ ] [enkidevs/curriculum/lua/lua-core/intro-to-lua-ii/installing-lua-i.md](https://app.enki.com/public/preview-insight/enkidevs/curriculum/lua/lua-core/intro-to-lua-ii/installing-lua-i.md?ref=885261a0c1b5c73d2c6127d272009604b479b7f5)
- [ ] [enkidevs/curriculum/lua/lua-core/intro-to-lua-ii/installing-lua-ii.md](https://app.enki.com/public/preview-insight/enkidevs/curriculum/lua/lua-core/intro-to-lua-ii/installing-lua-ii.md?ref=885261a0c1b5c73d2c6127d272009604b479b7f5)
- [ ] [enkidevs/curriculum/lua/lua-core/intro-to-lua-ii/installing-lua-iii.md](https://app.enki.com/public/preview-insight/enkidevs/curriculum/lua/lua-core/intro-to-lua-ii/installing-lua-iii.md?ref=885261a0c1b5c73d2c6127d272009604b479b7f5)
- [ ] [enkidevs/curriculum/lua/lua-core/intro-to-lua-ii/installing-lua-iv.md](https://app.enki.com/public/preview-insight/enkidevs/curriculum/lua/lua-core/intro-to-lua-ii/installing-lua-iv.md?ref=885261a0c1b5c73d2c6127d272009604b479b7f5)
- [ ] [enkidevs/curriculum/lua/lua-core/intro-to-lua/intro-code-lua.md](https://app.enki.com/public/preview-insight/enkidevs/curriculum/lua/lua-core/intro-to-lua/intro-code-lua.md?ref=885261a0c1b5c73d2c6127d272009604b479b7f5)
- [ ] [enkidevs/curriculum/lua/lua-core/intro-to-lua/what-to-expect.md](https://app.enki.com/public/preview-insight/enkidevs/curriculum/lua/lua-core/intro-to-lua/what-to-expect.md?ref=885261a0c1b5c73d2c6127d272009604b479b7f5)
- [ ] [enkidevs/curriculum/lua/lua-core/intro-to-lua/why-learn-lua.md](https://app.enki.com/public/preview-insight/enkidevs/curriculum/lua/lua-core/intro-to-lua/why-learn-lua.md?ref=885261a0c1b5c73d2c6127d272009604b479b7f5)

nem035 commented on lua/lua-core/intro-to-lua-ii/installing-lua-i.md:
```diff
@@ -6,13 +6,23 @@ type: normal
+Now hold your horses! Before you can start coding in Lua, **you need a way to code and test Lua language**.
 
-For the interpreter, you can easily find it on the download page of the [official Lua website](https://www.lua.org/download.html).
+For this course, coding on **an online IDE[1] is perfectly fine**! You can view online demo programs here: [live demo](https://www.lua.org/demo.html).
```
```suggestion
For this course, coding on **an online IDE[1] is perfectly fine**! You can view online demo programs [here](https://www.lua.org/demo.html).
```

nem035 commented on lua/lua-core/intro-to-lua/what-to-expect.md:
```diff
@@ -11,10 +11,11 @@ type: normal
-![rotating-block](https://img.enkipro.com/ec104dd401c8fe092d3950f5bde6ddb1.gif)
\ No newline at end of file
+> Learning Lua will help you build Roblox games and more, which is why we suggest it as your starting point.
+> However, if you'd like, you can skip Lua and jump into our [Roblox Studio course](https://app.enki.com/course/roblox-studio).
```
```suggestion
> If you'd like, you can skip Lua and jump into our [Roblox Studio course](https://app.enki.com/course/roblox-studio).

```

enki-hq commented:
### Insight

| Insight Name | Error Message |
|:-:|:-:|
| python-environment | <ul><li>`Install Python` pointing to `https://wiki.python.org/moin/BeginnersGuide/Download` is broken because `HTTP_503`</li></ul> |

> If you notice anything wrong with these rules contact content-team@enki.com.

junoocha commented:
New Changes:
- Added the commits from Nem
- For insight 2, I moved the note regarding the optionality of the next 3 insights to the top, as it should be the first thing a user should read before deciding
- Removed the mention of LuaDroid to keep the focus on Enki

junoocha commented:
> New Changes:
> 
> * Added the commits from Nem
> * For insight 2, I moved the note regarding the optionality of the next 3 insights to the top, as it should be the first thing a user should read before deciding
> * Removed the mention of LuaDroid to keep the focus on Enki

I might also consider removing most of the content in this workout, as it's extremely likely that users will be coding on Roblox to begin with, hence there's no need for this portion. IF the change does go through, I'll likely keep the online demo of the Lua Code, and then directing to the Roblox Studio insight where users can start scripting/coding.

junoocha commented:
Reverted the changes regarding deletion of files to properly archive them in a different PR. 

The plan is to completely remove most of the installation workout and move the first insight to the introduction workout. This first insight was edited to explain that the online IDE is enough for the course, and also directed them to the Roblox Studio intro to scripting workout if they'd like to code through Studio (as it comes with Lua installed)

Additionally, as the course has now switched to the use of an online IDE, there is no use for IO operations. This is furthered by the fact that these operations are deprecated in Roblox Studio. Thus, the two workouts that explore IO operations will also be archived.

enki-hq commented:
### Insight

| Insight Name | Error Message |
|:-:|:-:|
| installing-lua-i | Insight slug `installing-lua-i` is not unique in the database. |

> If you notice anything wrong with these rules contact content-team@enki.com.

Stefan-Stojanovic commented on lua/lua-core/intro-to-lua-ii/installing-lua-i.md:
```diff
@@ -6,13 +6,21 @@ type: normal
 ---
 
-# Installing the Interpreter I
+# Getting Starting to Code Lua
```
```suggestion
# Starting to Code in Lua
```

Or

```suggestion
# Getting Started With Lua Code
```

Stefan-Stojanovic commented on lua/lua-core/intro-to-lua-ii/installing-lua-iv.md:
```diff
@@ -0,0 +1,23 @@
+
+## Content
+
+As for text editors, there are many options to choose from, including **Sublime Text**, **Atom**, **Notepad++**, **VSCode** and more. 
```
```suggestion
As for text editors, there are many options to choose from, including **Sublime Text**, **Notepad++**, **VSCode** and more. 
```

Remove  **Atom** as it was deprecated last year or the year before that

junoocha replied:
Should be noted that insights 2, 3, 4 will be removed and insight 1 will be moved into the first workout, effectively deprecating the installation workout.

This will be carried out immediately after this PR merges through the arching Lua workout PR.

enki-hq commented:
### Insight

| Insight Name | Error Message |
|:-:|:-:|
| types-of-networks | <ul><li>`Computer Network Types` pointing to `https://study.com/academy/lesson/types-of-networks-lan-wan-wlan-man-san-pan-epn-vpn.html` is broken because `HTTP_403`</li></ul> |
| installing-lua-iv | <ul><li>`Download VSCode` pointing to `https://code.visualstudio.com/download` is broken because `HTTP_500`</li></ul> |

> If you notice anything wrong with these rules contact content-team@enki.com.

enki-hq commented:
### Insight

| Insight Name | Error Message |
|:-:|:-:|
| installing-lua-iv | <ul><li>`Download VSCode` pointing to `https://code.visualstudio.com/download` is broken because `HTTP_500`</li></ul> |

> If you notice anything wrong with these rules contact content-team@enki.com.

enki-hq commented:
### Insight

| Insight Name | Error Message |
|:-:|:-:|
| installing-lua-iv | <ul><li>`Download VSCode` pointing to `https://code.visualstudio.com/download` is broken because `HTTP_500`</li></ul> |

> If you notice anything wrong with these rules contact content-team@enki.com.
